src/list-ext.satyg

```diff
@@ -5,36 +5,30 @@
 @import: ord
 
 module List : sig
+  val eq : ('a Eq.t) implicit -> (('a list) Eq.t) implicit
+
   val null : 'a list -> bool
   val nil : 'a list
   val cons : 'a -> 'a list -> 'a list
+  val uncons : 'a list -> ('a * 'a list) option
   val map : ('a -> 'b) -> 'a list -> 'b list
   val mapi : (int -> 'a -> 'b) -> 'a list -> 'b list
   val iter : ('a -> unit) -> 'a list -> unit
   val iteri : (int -> 'a -> unit) -> 'a list -> unit
-  val fold-left : ('a -> 'b -> 'a) -> 'a -> 'b list -> 'a
-  val fold-lefti : (int -> 'a -> 'b -> 'a) -> 'a -> 'b list -> 'a
-  val fold-right : ('a -> 'b -> 'b) -> 'b -> 'a list -> 'b
   val filter : ('a -> bool) -> 'a list -> 'a list
-  val assoc : ('a -> 'a -> bool) -> 'a -> ('a * 'b) list -> 'b option
   val reverse : 'a list -> 'a list
   val append : 'a list -> 'a list -> 'a list
   val concat : ('a list) list -> 'a list
-  val fold-left-adjacent : ('a -> 'b -> 'b option -> 'b option -> 'a) -> 'a -> 'b list -> 'a
-  val map-adjacent : ('a -> 'a option -> 'a option -> 'b) -> 'a list -> 'b list
-  val mapi-adjacent : (int -> 'a -> 'a option -> 'a option -> 'b) -> 'a list -> 'b list
   val length : 'a list -> int
   val nth : int -> 'a list -> 'a option
-
-  val eq : ('a Eq.t) implicit -> (('a list) Eq.t) implicit
   val intersperse : 'a -> 'a list -> 'a list
   val find : ('a -> bool) -> 'a list -> 'a option
   val apply : 'a -> ('a -> 'b) list -> 'b list
   val take : int -> 'a list -> 'a list
   val drop : int -> 'a list -> 'a list
-  val takewhile : ('a -> bool) -> 'a list -> 'a list
-  val dropwhile : ('a -> bool) -> 'a list -> 'a list
-  val splitat : int -> 'a list -> ('a list * 'a list)
+  val take-while : ('a -> bool) -> 'a list -> 'a list
+  val drop-while : ('a -> bool) -> 'a list -> 'a list
+  val split-at : int -> 'a list -> ('a list * 'a list)
```

Comment:
kebab-case is preferred
(The old ones are moved below as they are obsoleted)

Reply:
![Lisplogo_alien_256](https://user-images.githubusercontent.com/540144/80679621-35f51c80-8af8-11ea-86d2-c4bf1418fcf5.png)

Reply:
haha